src/memtest.rs

```diff
@@ -545,3 +907,77 @@ where
     let str = String::deserialize(deserializer)?;
     Ok(anyhow!(str))
 }
+
+#[cfg(test)]
+mod test {
+    use {
+        super::{MemtestKind, MemtestOutcome, TestObserver},
+        std::{error::Error, fmt},
+    };
+
+    #[derive(Debug)]
+    struct IterationCounter {
+        expected_iter: Option<u64>,
+        completed_iter: u64,
+    }
+
+    // This is just a place holder, it will never be used.
```

Comment:
Ah... Luckily, there actually exists a specific Rust type in `std` for this exact scenario: [`std::convert::Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html). With it, you can write `fn foo() -> Result<String, Infallible>` and Rust understands that the `Err<E>` variant will never be constructed.